doc/lsst.pipe.base/creating-a-command-line-task.rst

```diff
@@ -0,0 +1,253 @@
+
+Reading and Writing Data
+========================
+
+The :ref:`run method <creating-a-command-line-task-intro>` typically receives a single data reference, as mentioned above.
+It read and writes data using this data reference (or the underlying butler, if necessary).
+
+.. _creating-a-command-line-task-dataset-types:
+
+Adding Dataset Types
+--------------------
+
+Every time you write a task that writes a new kind of data (a new "dataset type") you must tell the butler about it.
+Similarly, if you write a new task for which you want to save configuration and metadata (which is the case for most tasks that process data), you have to tell the butler about it.
+
+To add a dataset, edit the mapper configuration file for each observatory package on whose data the task can be run.
+If the task is of general interest (wanted for most or all observatory packages) then this process of updating all the mapper configuration files can be time consuming.
+
+There are plans to change how mappers are configured.
+But as of this writing, mapper configuration files are contained in the policy directory of the observatory package.
+For instance the configuration for the lsstSim mapper is defined in ``obs_lsstSim/policy/LsstSimMapper.paf``.
+
+.. _creating-a-command-line-task-persisting-config-and-metadata:
+
+Persisting Config and Metadata
+==============================
+
+Normally when you run a task you want the configuration for the task and the metadata generated by the task to be saved to the data repository.
+By default, this is done automatically, using dataset types:
+
+- ``_DefaultName_config`` for the configuration
+- ``_DefaultName_metadata`` for the metadata
+
+where ``_DefaultName`` is replaced with the value of the task's ``_DefaultName``, see :ref:`class variable <creating-a-task-class-variables>`.
+
+Whether you use these default dataset types or :ref:`customize the dataset types <creating-a-command-line-task-custom-config-and-metadata-types>`, you will have to :ref:`add dataset types <creating-a-command-line-task-dataset-types>` for the configuration and metadata.
+
+.. _creating-a-command-line-task-custom-config-and-metadata-types:
+
+Customizing Config and Metadata Dataset Types
+---------------------------------------------
+
+Occasionally the default dataset types for configuration and metadata are not sufficient.
+For instance in the case of the `pipe.tasks.makeSkyMap.MakeSkyMapTask` and various co-addition tasks, the co-add type must be part of the config and metadata dataset type name.
+To customize the dataset type of a task's config or metadata, define task methods ``_getConfigName`` and ``_getMetadataName`` to return the desired names.
+
+.. _creating-a-command-line-task-prevent-saving-config:
+
+Prevent Saving Config and Metadata
+----------------------------------
+
+For some tasks you may wish to not save config and metadata at all.
+This is appropriate for tasks that simply report information without saving data.
```

Comment:
What sort of tasks?  And how do they return information without saving metadata -- it sounds as if these are exactly the tasks that *need* to write metadata.

This may require an extension to the metadata system.